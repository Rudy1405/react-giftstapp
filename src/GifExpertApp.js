import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({defaultCategories=[]}) => { // se agrega el defaultCategories para dejar que se pueda enviar data con la cual renderear el componente, se uso para TESTING sino, solo quitalo y en use State un []

    //const categories = ["One Punch","SNK","cool cats"]
    const [categories, setCategories] = useState(defaultCategories);
   
    return (
        <>
            <h2 className="animate__animated animate__jackInTheBox animate__delay-1s"> Gif Detective App</h2>
            <AddCategory setCategories={setCategories}/>
            <hr className="animate__animated animate__zoomIn animate__delay-0.5s" />
            <ol>
                {
                    categories.map( (category,idx) => (
                        <GifGrid  // Aqui usamos los () en lugar de {} para denotar el return lo cual crea una instancia de GifGrid por cada enter en AddCategory
                            key = {category} 
                            category= {category}
                        />
                    )) 
                }
                
            </ol>
        </>
    )
}
