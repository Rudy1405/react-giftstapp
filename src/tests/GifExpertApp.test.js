import React from "react"
import "@testing-library/jest-dom";
import { shallow } from "enzyme"
import { GifExpertApp } from "../GifExpertApp";

describe('Testing in <GifExpertApp/>', () => {
    
    test('should render correctly', () => {
        let wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    /// el useState no se puede validar como yal... por ende evaluamos si se rendereo lo que el state altera
    test("Should show a list of categories", () =>{
        const categories = ["patas", "snk"];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);
    });

})
