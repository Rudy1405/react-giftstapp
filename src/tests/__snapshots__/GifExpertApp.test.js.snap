// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <GifExpertApp/> Should show a list of categories 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__jackInTheBox animate__delay-1s"
  >
     Gif Detective App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr
    className="animate__animated animate__zoomIn animate__delay-0.5s"
  />
  <ol>
    <GifGrid
      category="patas"
      key="patas"
    />
    <GifGrid
      category="snk"
      key="snk"
    />
  </ol>
</Fragment>
`;

exports[`Testing in <GifExpertApp/> should render correctly 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__jackInTheBox animate__delay-1s"
  >
     Gif Detective App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr
    className="animate__animated animate__zoomIn animate__delay-0.5s"
  />
  <ol />
</Fragment>
`;
